src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -284,6 +284,20 @@ public boolean isOnGround() {
         return this.entityMeta;
     }
 
+    /**
+     * Do a batch edit of this entity's metadata.
+     */
+    public <TMeta extends EntityMeta> void editEntityMeta(Class<TMeta> metaClass, Consumer<TMeta> editor) {
+        entityMeta.setNotifyAboutChanges(false);
+        try {
+            TMeta casted = metaClass.cast(entityMeta);
+            editor.accept(casted);
+        } catch (Throwable t) {
+            throw new RuntimeException("Error editing entity " + id + " " + entityType.name() + " meta", t);
+        }
+        entityMeta.setNotifyAboutChanges(true);
```

Comment:
this should happen in a `finally` block. If an exception occurs it will leave setNotifyAboutChanges disables.